### ISSUE TITLE
When `dracut-install` fails, kiwi continues

### ISSUE THREAD

**Issue by ggardet** [OSInside/kiwi#2397]
## Problem description

When `dracut-install` fails, kiwi continues silently. Below, there is a log where a kernel module (`pcie_qcom`) was not found. All the kernel modules listed _after_ this unknown module are silently ignored. This resulted in a broken image.
I think kiwi should stop at this point and return an error.

```
[  310s] dracut-install: Failed to find module 'pcie_qcom'
[  310s] dracut[E]: FAILED:  /usr/lib/dracut/dracut-install -D /var/tmp/dracut.w1jxAo/initramfs -N ^i2o_scsi$ --kerneldir /lib/modules/6.6.2-1-default/ -m gpio_regulator virtio_gpu isp1760 qcom-rpmh-regulator qcom-ipcc pinctrl-sc8280xp pdr_interface phy_qcom_snps_femto_v2 i2c_hid llcc_qcom xhci_hcd dwc3_qcom usbcore xhci_plat_hcd dwc3 udc_core pmic_glink_altmode ulpi qcom_spmi_pmic regmap_spmi rc_core rtc_pm8xxx drm_display_helper pinctrl_spmi_gpio drm_dp_aux_bus qcom_vadc_common gpu_sched qcom_spmi_pmic phy_qcom_qmp_combo typec clk-rpmh dispcc-sc8280xp gcc-sc8280xp gpucc-sc8280xp nvmem_qcom-spmi-sdam qcom_hwspinlock qcom_q6v5 qcom_q6v5_pas qnoc-sc8280xp pmic_glink pmic_glink_altmode smp2p spmi-pmic-arb leds-qcom-lpg nvme phy_qcom_qmp_pcie pcie_qcom i2c_hid_of i2c_qcom_geni leds-qcom-lpg pwm_bl qrtr pmic_glink_altmode gpio_sbu_mux phy_qcom_qmp_combo panel-edp msm phy_qcom_edp
[  310s] dracut[I]: *** Installing kernel module dependencies ***
[  310s] dracut[I]: *** Installing kernel module dependencies done ***
[  310s] dracut[I]: *** Resolving executable dependencies ***
[  310s] dracut[I]: *** Resolving executable dependencies done ***
[  310s] dracut[I]: *** Hardlinking files ***
[  310s] dracut[D]: Mode:                     real
[  310s] dracut[D]: Method:                   memcmp
[  310s] dracut[D]: Files:                    3928
[  310s] dracut[D]: Linked:                   4 files
[  310s] dracut[D]: Compared:                 0 xattrs
[  310s] dracut[D]: Compared:                 554 files
[  310s] dracut[D]: Saved:                    1.18 MiB
[  310s] dracut[D]: Duration:                 0.029405 seconds
[  310s] dracut[I]: *** Hardlinking files done ***
[  310s] dracut[I]: *** Store current command line parameters ***
[  310s] dracut[I]: *** Stripping files ***
[  310s] dracut[I]: *** Stripping files done ***
[  310s] dracut[I]: *** Creating image file '/openSUSE-Tumbleweed-ARM-JeOS-efi.aarch64-2023.11.21.initrd' ***
[  310s] dracut[D]: Image: /var/tmp/dracut.w1jxAo/initramfs.img: 71M
[  310s] dracut[D]: ========================================================================
```

## Expected behaviour
kiwi should stop when dracut-install fails and return an error.

## Steps to reproduce the behaviour

Build an image with additional (unknown) kernel modules.

## OS and Software information

* KIWI version: 9.25.17
* Operating system host version: openSUSE Tumbleweed
* Operating system target version: openSUSE Tumbleweed
* Open Build Service version (N/A if not using OBS): build.opensuse.org
* Koji version (N/A if not using Koji): N/A

**Comment by schaefi** [comment]
Well the thing is, dracut itself does not report that it has failed. The exit code is 0 which is the reason why kiwi continues. We had discussed this in the past and we would need to analyse the dracut output and make a decision based on that if it was successful or not. As we thought this would be a pretty unstable and also flaky way to determine success or failure we decided to call dracut such that it prints all information it has such that users can at least look at the build log to identify issues as you did

Personally I think it's the responsibility of the tool itself to decide per exit code if that was a successful call or not, it's imho not a kiwi responsibility to judge that

Thoughts ?
